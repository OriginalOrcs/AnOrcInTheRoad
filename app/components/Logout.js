import React from 'react';
import { Dimensions, View, Text, StyleSheet } from 'react-native';
import { Font } from 'exponent';

let { height, width } = Dimensions.get('window')

class Logout extends React.Component {
	render(){
		return (
			<View style={styles.btn}>
			  <Text style={styles.btnText} onPress={this.props.onLogout}>Logout</Text>
			</View>
		);
	}
}


const styles = StyleSheet.create({
  btn: {
    backgroundColor: '#b9d3c2',
    width: 150,
    height: 40,
    borderColor: '#336A73',
    borderWidth: 2,
    marginTop: 100,
    marginLeft: 20,
    marginRight: 20,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
  },
  btnText: {
    color: 'white',
    fontSize: 25,
    fontWeight: '600',
    ...Font.style('livingst'),
    borderRadius: 10,
  },
});

export default Logout;
