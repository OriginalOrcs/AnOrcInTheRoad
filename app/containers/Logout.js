import { connect } from 'react-redux';
import { userLogout } from '../actions/actions';
import Logout from '../components/Logout';
import socket from '../main';
import Exponent from 'exponent';


const mapStateToProps = (state) => {
  return {
    user: state.user,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    onLogout: () => {
    	console.log('DISPATCH LOGOUT')
      dispatch(userLogout());
      Exponent.WebBrowser.openBrowserAsync('https://originalorcs.auth0.com/v2/logout');
      Exponent.WebBrowser.dismissBrowser();
      Exponent.Util.reload();
    },
  };
};

const LogoutButton = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Logout);

export default LogoutButton;