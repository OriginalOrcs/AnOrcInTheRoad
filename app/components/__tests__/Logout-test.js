import 'react-native';
import React from 'react';
import Logout from '../Logout';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Logout />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
