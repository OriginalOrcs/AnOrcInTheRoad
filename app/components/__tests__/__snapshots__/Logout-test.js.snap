exports[`test renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={undefined}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#05A5D1",
        "flexDirection": "row",
        "fontFamily": "ExponentFont-[object Object]-luminari",
        "fontSize": 20,
        "fontWeight": "300",
        "padding": 20,
        "width": 375,
      }
    }>
    Logout
  </Text>
</View>
`;
